client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -37,16 +38,14 @@ export default React.createClass( {
 	render() {
 		let title;
 
-		if ( this.props.postViewsList.response.post && this.props.postViewsList.response.post.post_title ) {
-			title = this.translate( 'Stats for %(posttitle)s', {
-				comment: 'Title of the individual post stats page.',
-				args: {
-					posttitle: this.props.postViewsList.response.post.post_title
-				}
-			} );
-		}
+		const post = this.props.postViewsList.response.post;
+		const postOnRecord = post && post.post_title !== null;
 
-		if ( this.props.postViewsList.isError() ) {
+		if ( postOnRecord && ! this.props.postViewsList.isError() ) {
+			if ( typeof post.post_title === 'string' && post.post_title.length ) {
+				title = <Emojify>{ post.post_title }</Emojify>;
+			}
+		} else {
```

Comment:
> What do you think about defaulting title to the "We don't have that post on record yet" in line 39?

It's a limited case, but for posts with an empty post title, defaulting title to the "not-on-record" label would mean needlessly calling the translate function just to turn around and set it to an empty string for posts w/ empty titles.